src/bin/graph_msgs.rs

```diff
@@ -0,0 +1,224 @@
+            Output file row format:\n\n\
+            step_number [network_size] [queue_size] [total_messages_sent] [avg_messages_sent] \
+            [max_messages_sent_per_node]")
+        .arg(Arg::with_name("output")
+                 .short("o")
+                 .long("output")
+                 .value_name("FILE")
+                 .help("The name for the output file."))
+        .arg(Arg::with_name("include_network_size")
+                 .short("n")
+                 .long("network-size")
+                 .takes_value(false)
+                 .help("Include the network size in the output"))
+        .arg(Arg::with_name("include_queue_size")
+                 .short("q")
+                 .long("queue-size")
+                 .takes_value(false)
+                 .help("Include the queue size in the output"))
+        .arg(Arg::with_name("include_total_sent")
+                 .short("t")
+                 .long("total-sent")
+                 .takes_value(false)
+                 .help("Include the total number of messages sent in the output"))
+        .arg(Arg::with_name("include_avg_sent")
+                 .short("a")
+                 .long("average-sent")
+                 .takes_value(false)
+                 .help("Include the average number of messages sent per node in the output"))
+        .arg(Arg::with_name("include_max_sent")
+                 .short("m")
+                 .long("max-sent")
+                 .takes_value(false)
+                 .help("Include the maximum number of messages sent per node in the output"))
+        .arg(Arg::with_name("plot")
+                 .short("p")
+                 .long("plot")
+                 .value_name("PLOT")
+                 .help("Plot the graph using gnuplot to the given file"))
+        .arg(Arg::with_name("INPUT")
+                 .help("Sets the input file to use")
+                 .required(true)
+                 .index(1))
+        .get_matches();
+    let input = matches.value_of("INPUT").unwrap();
+    let output = matches.value_of("output").unwrap_or("output.dot");
+    let network_size = matches.is_present("include_network_size");
+    let queue_size = matches.is_present("include_queue_size");
+    let total_sent = matches.is_present("include_total_sent");
+    let avg_sent = matches.is_present("include_avg_sent");
+    let max_sent = matches.is_present("include_max_sent");
+    let plot = matches.value_of("plot");
+    //let mut blocks = BTreeSet::new();
+    let mut sent_msgs = BTreeMap::new();
+    let mut node_names = BTreeSet::new();
+    let mut result = Vec::new();
+    let mut msgs_in_queue = 0;
+
+    let file = File::open(input).unwrap();
+    let log_iter = LogIterator::new(file);
+
+    println!("Reading log...");
+    for data in log_iter {
+        match data {
+            /*LogData::VoteAgreement(_, block_from, block_to) => {
+                blocks.insert(block_from);
+                blocks.insert(block_to);
+            }*/
```

Comment:
I left it in there because I didn't want to rewrite it from scratch if we decided that we still want to count the valid blocks per prefix somehow - but if we don't, then it can be removed, that's true.